Fix headers check

Y0ba opened the pull request.

SthephanShinkufag commented on src/modules/Ajax.js:
```diff
@@ -227,9 +227,15 @@ const AjaxCache = {
 		let headers = 'getAllResponseHeaders' in xhr ? xhr.getAllResponseHeaders() : xhr.responseHeaders;
 		headers = headers ? /* usual xhr */ headers.split('\r\n') : /* fetch */ xhr.headers;
-		for(let header in headers) {
+		for(let header of headers) {
```
https://github.com/SthephanShinkufag/Dollchan-Extension-Tools/commit/93cb4fb6e5374431bed4c0e1d551872fc5c9c389

SthephanShinkufag replied:
https://dscript.me/de/res/1.html#763
https://dscript.me/de/res/1.html#796

Y0ba replied:
`in` перебирает индексы массива, а не значения. На выходе были `1`, `2`, `3` и т.д.

SthephanShinkufag replied:
Тем не менее, это фиксило баг. Проверь firefox 60

Y0ba replied:
Тогда вместо `of` можно использовать `in`, но в таком виде
```js
for(const idx in headers) {
    let header = headers[idx];
    ...
}
```

Y0ba replied:
Оно фиксило кривую работу с итераторами, но ломало кэш. Тогда сделаю с `in`, раз `of` не работает. Может тогда вообще `of` везде поубирать?

SthephanShinkufag replied:
Да, не заметил, что сделал глупость.
Странно, почему там `of` в некролисе не работает....

Y0ba commented:
Тогда заменить все остальные `of` на `in` или оставить как есть?

SthephanShinkufag commented:
А можно перебор итерируемых объектов без опаски менять на `for .. in`? Тогда в чём сакральный смысл `for .. of`?

Y0ba commented:
`for of` использует универсальный протокол итератора. Можно впихнуть итератор в свой объект и использовать `for of` с ним. `for in` работает вроде только с массивами и объектами.

Y0ba commented:
Кстати, скрипт использует протокол итераторов для простой итерации постов:
https://github.com/SthephanShinkufag/Dollchan-Extension-Tools/blob/df38c56909130f3093be793e6b32883b167cc29d/src/modules/DelForm.js#L64-L76

Тут уже `for of` так просто не заменишь.

SthephanShinkufag commented:
Лучше лишнего не менять. Если на остальное некроюзеры не жалуются, то лучше и не трогать.

Y0ba commented:
Хотя я кажется понял в чём дело. Некролиса не может использовать итератор при получении хеадеров из `fetch`. Можно оставить `for of`, но поменять
```js
headers = headers ? /* usual xhr */ headers.split('\r\n') : /* fetch */ xhr.headers;
```
на
```js
headers = headers ? /* usual xhr */ headers.split('\r\n') : /* fetch */ Array.from(xhr.headers);
```

SthephanShinkufag commented:
Главное, затесть в 60 некролисе

https://mozilla-firefox-portable.ru.uptodown.com/windows/versions

SthephanShinkufag commented:
И да, если будешь проверять, fetch используется только на тиниброд-досках, а также при кроссдоменных запросах при использовании расширений вместо скриптменеджера.

Y0ba commented:
Я думаю лучше оставить как есть. `Array.from` только оверхеда добавит для некоторых юзеров. Раз так работает, то пусть и будет.
